### PR TITLE
feat: udpate test codes ✅

### PR DESCRIPTION
**Pull request by dusunax**
### Overview
feat: Update test code, replace mocking values with the real responses of functions

### Why Change?
Reflect feedback from **Thursday mentoring session**. T9H.

### What Changed
- 🗑️ Removed mocking values.
- 🗑️ Removed unnecessary test code.

### Test Covarage >> about the same 🐯
image

### CODE REVIEW

File: src/command/colors.command.spec.ts [hunk 1]
```diff
@@ -1,18 +1,8 @@
...
-import * as colorHelper from '../helper/color.helper';
 import { colors as colorsCommand } from './colors.command';
 
-jest.mock('../helper/color.helper', () => ({
-  listColors: jest
-    .fn()
-    .mockReturnValue(
-      '\x1B[30mblack\x1B[0m, \x1B[31mred\x1B[0m, \x1B[32mgreen\x1B[0m, \x1B[33myellow\x1B[0m, \x1B[34mblue\x1B[0m, \x1B[35mmagenta\x1B[0m, \x1B[36mcyan\x1B[0m, \x1B[37mwhite\x1B[0m, \x1B[91mbrightRed\x1B[0m, \x1B[92mbrightGreen\x1B[0m, \x1B[93mbrightYellow\x1B[0m, \x1B[94mbrightBlue\x1B[0m, \x1B[95mbrightMagenta\x1B[0m, \x1B[96mbrightCyan\x1B[0m, \x1B[97mbrightWhite\x1B[0m\n',
-    ),
-}));
```
Comment (tolluset) [inline src/command/colors.command.spec.ts:11]: We can consider adding a snapshots test about the checking diff of the color list.

https://jestjs.io/docs/snapshot-testing
https://www.daleseo.com/jest-snapshot/

**Review by tolluset — APPROVED** [review]
LGTM! 👍 Nice refactoring!

File: src/command/colors.command.spec.ts (hunk 1 above)
Comment (dusunax) [reply src/command/colors.command.spec.ts:11]: Thank you for letting me know😀
I added snapshot test to `color.helper.spec.ts`. 
```tsx
test('listColors should return a string with all colors', () => {
  const result = listColors();
  const colorKeys = Object.keys(COLOR)
    .filter((e) => e !== 'reset')
    .map((e) => `${COLOR[e]}${e}${COLOR.reset}`)
    .join(', ');

  expect(result).toMatch(colorKeys); // 1️⃣ 
  expect(result).toMatchSnapshot(); // 2️⃣ snapshot (check previous result on `./__snapshots__/color.helper.spec.ts.snap`)
});
```
image

**I have a question,** 
When testing snapshots, is it preferable to test the colorKeys on 1️⃣, _only when a snapshot does not exist??_
Is it not redundant? to conduct both (1)snapshot and (2)functionality tests every time?🤔
I haven't tried writing snapshot test code before, so I'm not sure about it.

Comment (tolluset) [reply src/command/colors.command.spec.ts:11]: > When testing snapshots, is it preferable to test the colorKeys on 1️⃣, only when a snapshot does not exist??

Advantage of snapshots test is not have to helper function for testing.

We can test by 1️⃣, but snapshots test is not dependent to color keys data structure. E.g. If colorKeys data structure is changed to another, the test will fail, and the helper function that for testing have to fix it. But, in snapshots test we have to just update test.

Comment (dusunax) [reply src/command/colors.command.spec.ts:11]: > thank you for the comment✨

👉 I organized the test code for `src/helper/color.helper.ts` as follows:

(1) Verify if the output of the random color is in ANSI color format.
(2-1) Check if the color list is not null.
(2-2) Confirm if the displayed color list matches the previous snapshot. 👈

## update snapshot 😄

- make diff on COLORS

image

- update!

image
image

## and also 

found the issue of a **type error** occurring each time key&value is added to the `COLOR` array,
I moved the `COLOR` constant to `src/type/color.type.ts` & change `ColorType`like below (and change COLOR to COLORS)

```tsx
export type ColorType = keyof typeof COLORS;
```
(If we need conventions for constants later, we can move constants to another location)

**Review by BeomyeonAndrewKim — APPROVED** [review]
Awesome!

**Pull request merged**